src/locales/en.ts

```diff
@@ -71,10 +71,18 @@ const errorMap: ZodErrorMap = (issue, _ctx) => {
         message = `Array must contain ${
           issue.exact ? "exactly" : issue.inclusive ? `at least` : `more than`
         } ${issue.minimum} element(s)`;
-      else if (issue.type === "string")
-        message = `String must contain ${
+      else if (issue.type === "string"){
+        if (issue.minimum<=1){
+          message = `String must contain ${
+            issue.exact ? "exactly" : issue.inclusive ? `at least` : `over`
+          } ${issue.minimum} character`
```

Comment:
You could just do:

```ts
message = `String must contain ${
            issue.exact ? "exactly" : issue.inclusive ? `at least` : `over`
          } ${issue.minimum} character${issue.minimum > 1 ? "(s)" : ""`;
```

Reply:
thanks for the explanation.

Reply:
should i do the changes or its done.

Reply:
Make the change, duplication is never nice.

Reply:
so i just overwrite my code with yours??